src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -37,13 +37,25 @@ private bool ValidateReaderType(string readerType)
 #pragma warning disable SYSLIB0011
         private object ReadBinaryFormattedObject()
         {
-            BinaryFormattedObject binaryFormattedObject = new BinaryFormattedObject(_store.BaseStream,
-                new BinaryFormattedObject.Options()
+            long position = _store.BaseStream.CanSeek ? _store.BaseStream.Position : -1;
+
+            try
+            {
+                BinaryFormattedObject binaryFormattedObject = new BinaryFormattedObject(_store.BaseStream);
```

Comment:
One of the other reasons things deserialization could fail is because the stream is not seekable, right? This doesn't come up with embedded resources, afaik.